test/request/component-models/LoggingComponentModel.spec.ts

```diff
@@ -10,6 +10,307 @@ import * as chai from 'chai';
 
 const should = chai.should();
 
+describe('LogMessageModel', () => {
```

Comment:
Probably should add a test case for where spans run back-to-back.  e.g., [0,3], [3, 6].   Probably also want to add a test case for when a span is at the end of the string (e.g., [0,3] for a string of length 4.